frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-design/experiment-design.component.ts

```diff
@@ -301,6 +306,7 @@ export class ExperimentDesignComponent implements OnInit, OnChanges, OnDestroy {
       this.previousAssignmentWeightValues.splice(groupIndex, 1);  
       this.applyEqualWeight();
     }
+    this.isRowRemoved = true;
```

Comment:
what is this new variable for (how does it relate to speedbump?).

If it is necessary to have this, it appears to be initialized as false and basically immediately set to true, perhaps it can just be initialized as true.

Reply:
I have used ng-form.dirty to detect changes in form but it wasn't detecting deletion of rows in tables so I have created this variable and set is as false but if any row gets deleted then it will be true and added that variable in if condition with ng-form.dirty. 
![image](https://user-images.githubusercontent.com/50392803/198647799-1a668fe7-2923-4703-ae3c-5971918fad33.png)